### ISSUE TITLE
Efficient Weight Masking

### ISSUE THREAD

**Issue by KaleabTessera** [pytorch/xla#2558]
## ❓ Questions and Help

I am masking my weights of my linear and conv layers, in each forward pass as follows: 
```
def mask_weights(self):
        if self.mask is not None:
            with torch.no_grad():
                self.weight.data = torch.mul(self.weight.data,self.mask)

def forward(self, input):
        self.mask_weights()
        f = F.linear(input, self.weight, self.bias)
        return f
```

This is pretty slow using tpu. Is there a more efficient way to do this?

**Comment by KaleabTessera** [comment]
So the issue was that my masks were on cpu and not a tpu device.
